sdk/trace/trace_test.go

```diff
@@ -870,6 +872,119 @@ func TestCustomStartEndTime(t *testing.T) {
 	}
 }
 
+func TestRecordError(t *testing.T) {
+	te := &testExporter{}
+	tp, _ := NewProvider(WithSyncer(te))
+	span := startSpan(tp, "SpanEnd")
+
+	testErr := errors.New("test error")
+	errTime := time.Now()
+	span.RecordError(context.Background(), testErr,
+		apitrace.WithErrorTime(errTime),
+	)
+
+	got, err := endSpan(te, span)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := &export.SpanData{
+		SpanContext: core.SpanContext{
+			TraceID:    tid,
+			TraceFlags: 0x1,
+		},
+		ParentSpanID:    sid,
+		Name:            "span0",
+		SpanKind:        apitrace.SpanKindInternal,
+		HasRemoteParent: true,
+		MessageEvents: []export.Event{
+			{
+				Name: "error",
+				Time: errTime,
+				Attributes: []core.KeyValue{
+					core.Key("error.type").String(reflect.TypeOf(testErr).String()),
+					core.Key("error.message").String(testErr.Error()),
```

Comment:
Should we hardcode those fields? Currently these lines are basically copies of the SDK implementation.

Reply:
I moved the `core.Key()` values into unexported package constants and used them in the implementation and tests.  If that doesn't address your concern, please let me know what you're looking for.

Reply:
Ah, sorry for not being clear. I basically meant that there should be something like:

```go
core.Key("error.type").String("*errors.errorString"),
core.Key("error.message").String("test error"),
```

That way we avoid copying what SDK is doing in the test. This particular hardcoding kinda stinks, because it relies on some implementation detail of the errors package. So I was wondering if we could our own test implementation of the error interface and use that in test:

```go

type testError string

var _ error = testError("")

func newTestError(s string) error {
	return (testError)(s)
}

func (e testError) Error() string {
	return (string)(e)
}

…

err := newTestError("test error")

…

core.Key("error.type").String("go.opentelementry.go/otel/sdk/trace.testError"),
core.Key("error.message").String("test error"),
```

Reply:
Ok, I get what you mean.  Update inbound.